### ISSUE TITLE
[Bug]: error TS2304: Cannot find name 'step'

### ISSUE THREAD

**Issue by rfreydi** [storybookjs/storybook#26549]
### Describe the bug

As described in the documentation [Group interactions with the step function](https://storybook.js.org/docs/writing-tests/interaction-testing#group-interactions-with-the-step-function)

I can't group interactions together as typescript can not find name 'step'.

You can find in the reproduction repository a fresh installation of storybook.new. I just added a "play" function in ButtonStories with a step inside it.

Thank you in advance for your time !

### To Reproduce

https://stackblitz.com/edit/cannot-find-name-step?file=src%2Fstories%2Fbutton.stories.ts

### System

```bash
Storybook Environment Info:

  System:
    OS: Linux 5.0 undefined
    CPU: (8) x64 Intel(R) Core(TM) i9-9880H CPU @ 2.30GHz
    Shell: 1.0 - /bin/jsh
  Binaries:
    Node: 18.18.0 - /usr/local/bin/node
    yarn: 1.22.19 - /usr/local/bin/yarn
    npm: 10.2.3 - /usr/local/bin/npm <----- active
    pnp,: 8.15.3 - /usr/local/bin/pnpm
  npmPackages:
    @storybook/addon-docs: ^8.1.0-alpha.2 => 8.1.0-alpha.2
    @storybook/addon-essentials: ^8.1.0-alpha.2 => 8.1.0-alpha.2
    @storybook/addon-interactions: ^8.1.0-alpha.2 => 8.1.0-alpha.2
    @storybook/addon-links: ^8.1.0-alpha.2 => 8.1.0-alpha.2
    @storybook/angular: ^8.1.0-alpha.2 => 8.1.0-alpha.2
    @storybook/blocks: ^8.1.0-alpha.2 => 8.1.0-alpha.2
    @storybook/test: ^8.1.0-alpha.2 => 8.1.0-alpha.2
    storybook: ^8.1.0-alpha.2 => 8.1.0-alpha.2
```

**Comment by kasperpeulen** [comment]
You have to get up from the play `context`!



![Image](https://github.com/storybookjs/storybook/assets/1035299/75771bf1-7d84-44c3-baf2-3d820ec632f5)

**Comment by rfreydi** [comment]
Hi @kasperpeulen ! 

I don't understand your answer or I'm missing something.

If you open the reproduction provided in [stackblitz](https://stackblitz.com/edit/cannot-find-name-step?file=src%2Fstories%2Fbutton.stories.ts) it'll perform automatically a `npm install && yarn storybook` and throw an error.

Even if you run manually `npm run storybook` (because mixin npm and yarn may be error prone), you still get the same error.

The only difference I see between the reproduction code and your screen is that you added an `await` in front of the step function. I tried to do so but I still have the same error.

```
info => Starting manager..
info => Starting preview..
info Addon-docs: using MDX3
info => Using implicit CSS loaders
info => Using angular browser target options from "angular-latest:build"
info => Using angular project with "tsConfig:/home/projects/cannot-find-name-step/.storybook/tsconfig.json"
info => Using default Webpack5 setup
<i> [webpack-dev-middleware] wait until bundle finished
ERROR in src/stories/button.stories.ts:31:11 - error TS2304: Cannot find name 'step'.

31     await step('test', async () => {
             ~~~~

preview compiled with 1 error
WARN DefinePlugin
WARN Conflicting values for 'process.env.NODE_ENV'
=> Failed to build the preview
99% end closing watch compilationWARN Force closed preview build
SB_BUILDER-WEBPACK5_0003 (WebpackCompilationError): There were problems when compiling your code with Webpack.
Run Storybook with --debug-webpack for more information.
    at new StorybookError (./node_modules/@storybook/core-events/dist/errors/server-errors.js:6:2427)
    at new WebpackCompilationError (./node_modules/@storybook/core-events/dist/errors/server-errors.js:40:411)
    at starter (./node_modules/@storybook/builder-webpack5/dist/index.js:1:8163)
    at Module.start (./node_modules/@storybook/builder-webpack5/dist/index.js:1:10158)
    
Broken build, fix the error above.
You may need to refresh the browser.
```

**Comment by shilman** [comment]
@rfreydi 

Change L28:

```diff
- play: async ({ canvasElement }) => {
+ play: async ({ canvasElement, step }) => {
```
